src/node.js

```diff
@@ -253,6 +253,13 @@
             // be ok to abort on an uncaught exception at this point
             process._emittingTopLevelDomainError = true;
             caught = domain.emit('error', er);
+
+            // Exit all domains on the stack.  Uncaught exceptions end the
+            // current tick and no domains should be left on the stack
+            // between ticks.
+            var domainModule = NativeModule.require('domain');
+            domainStack.length = 0;
+            domainModule.active = process.domain = null;
```

Comment:
Certainly. This is an exact copy https://github.com/dohse/node/blob/clear-domain-stack/src/node.js#L279-L284 - it might be better to unify these two sites.